static/widgets/history-widget.ts

```diff
@@ -126,56 +87,29 @@ export class HistoryWidget {
         );
     }
 
-    private hideRadiosAndSetDiff() {
-        const root = unwrap(this.modal).find('.historiccode');
-        const items = root.find('li:not(.template)');
+    private getContent(src: EditorSource[]) {
+        return src.map(val => `/****** ${val.lang} ******/\n${val.source}`).join('\n');
+    }
```

Comment:
Maybe not the best to put it here? I assume it's only for display but it's probably best displayed either above, or not using a hacky "comment"

Reply:
This isn't my addition - you can see this added header when viewing diffs in the history pane in the live site.
I imagine it can be useful when there are multiple open editors with different languages.  But I knowingly ignored this case when adding syntax highlighting - instead I just took the language from the first editor. (I strongly believe it's a net positive)

Reply:
The original is here:
https://github.com/compiler-explorer/compiler-explorer/blob/7cecdad70dbe4f4142d21e826839c818811bf35e/static/widgets/history-widget.ts#L52

Reply:
You can have multiple languages in a history entry, so it serves as both a language indication as a separator. A constant might be nice, but there's not much other alternatives than doing this.